src/main/webapp/app/shared/learning-goal-selection/learning-goal-selection.component.ts

```diff
@@ -17,13 +17,16 @@ import { CourseStorageService } from 'app/course/manage/course-storage.service';
         },
     ],
 })
-export class LearningGoalSelectionComponent implements OnInit, ControlValueAccessor {
+export class LearningGoalSelectionComponent implements OnInit, OnChanges, ControlValueAccessor {
     @Input() labelName: string;
     @Input() labelTooltip: string;
     @Input() value: any;
     @Input() disabled: boolean;
     @Input() error: boolean;
     @Input() learningGoals: LearningGoal[];
+    nonOptionalLearningGoals: LearningGoal[];
+    selectableLearningGoals: LearningGoal[];
+    @Input() includeOptionals: boolean;
```

Comment:
You can think about giving this a default value to maybe be able to remove all instances where you simply set it to true. You can also leave it like that, so a competency cannot be made optional by accident when there is a bug

Reply:
I want to force Devs to set the `includeOptionals` flag. This way they are forced to decide actively. Otherwise, Devs might overlook this and accidentally allow linking to optional competencies.